### PR TITLE
Houdini: Expose Extract Active View Thumbnail to settings

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Exposes the Extractor that takes a quick screenshot of the active view to settings to allow disabling/enabling it.

## Additional info

Unfortunately we've been having some crashes with Houdini on this logic on a very heavy scene with heavy VDBs - unsure why and it's been inconsistent unfortunately, making it hard to debug and detect whether there's a way to report a bug to sidefx and have them fix Houdini.

This change now allows us to disable it for that project.

## Testing notes:

1. Tweak the settings
2. The Extract Active View Thumbnail should behave accordingly

### CODE REVIEW

File: server_addon/houdini/client/ayon_houdini/plugins/publish/extract_active_view_thumbnail.py [hunk 1]
```diff
@@ -17,6 +20,9 @@ class ExtractActiveViewThumbnail(plugin.HoudiniExtractorPlugin):
     families = ["workfile"]
 
     def process(self, instance):
+        if not self.is_active(instance):
```
Comment (moonyuet) [inline server_addon/houdini/client/ayon_houdini/plugins/publish/extract_active_view_thumbnail.py:23]:
```suggestion
        if not self.is_active(instance.data):
```

**Review by moonyuet — CHANGES_REQUESTED** [review]
Hit the error as we can't get the attribute from `Instance` object, we should get it from instance.data instead
image

**Review by moonyuet — APPROVED** [review]

**Comment by moonyuet** [comment]
@BigRoy please resolve the conflict and it can be possibly merged.

File: server_addon/houdini/server/settings/publish.py [hunk 2]
```diff
@@ -77,25 +77,29 @@ class PublishPluginsModel(BaseSettingsModel):
...
-        default_factory=BasicValidateModel,
+    ValidateUnrealStaticMeshName: BasicEnabledStatesModel = SettingsField(
+        default_factory=BasicEnabledStatesModel,
         title="Validate Unreal Static Mesh Name.")
     ValidateWorkfilePaths: ValidateWorkfilePathsModel = SettingsField(
         default_factory=ValidateWorkfilePathsModel,
         title="Validate workfile paths settings.")
+    ExtractActiveViewThumbnail: BasicEnabledStatesModel = SettingsField(
+        default_factory=BasicEnabledStatesModel,
+        title="Extract Active View Thumbnail"
```
Comment (MustafaJafar) [inline server_addon/houdini/server/settings/publish.py:101]: could we add a section
```suggestion
        title="Extract Active View Thumbnail",
        section="Extractors"
```
how my suggestion looks on my side
image

**Comment by BigRoy** [comment]
Thanks!

**Comment by BigRoy** [comment]
Ah - just bumped the addon version _just after_ haha. No worries, we'll just towards getting another PR merged asap with a version bump - this won't break things for now.

**Comment by moonyuet** [comment]
> Ah - just bumped the addon version _just after_ haha. No worries, we'll just towards getting another PR merged asap with a version bump - this won't break things for now.

I admit that it is getting a bit confused when you used with different Houdini PR and there is slightly different addon version.

**Pull request merged**